src/app/extensions/product-notifications/store/product-notification/product-notification.selectors.ts

```diff
@@ -0,0 +1,31 @@
+import { createSelector } from '@ngrx/store';
+
+import {
+  ProductNotification,
+  ProductNotificationType,
+} from '../../models/product-notification/product-notification.model';
+import { getProductNotificationsState } from '../product-notifications-store';
+
+import { initialState, productNotificationAdapter } from './product-notification.reducer';
+
+const getProductNotificationState = createSelector(getProductNotificationsState, state =>
+  state ? state.productNotifications : initialState
+);
+
+export const getProductNotificationsLoading = createSelector(getProductNotificationState, state => state.loading);
+
+export const getProductNotificationsError = createSelector(getProductNotificationState, state => state.error);
+
+const { selectAll } = productNotificationAdapter.getSelectors(getProductNotificationState);
+
+export const getAllProductNotifications = selectAll;
```

Comment:
this selector is not needed - you don't have to export it because it is never used and for internal purposes you can use selectAll

Reply:
Example taken from projects\requisition-management\src\app\store\requisitions\requisitions.selectors.spec.ts